hirbod/react-native-volume-manager
# [iOS] Add `initialQuery` param to `useSilentSwitch` and `addSilentListener`

# Why

Current API shape does not expose any mechanism to check the initial state of the `Ring/mute switch` on iOS. I need to ignore the very first meaningful status report of `Ring/mute switch` state (`true/false`) in React-friendly way. Therefore, I suggest API change and exposing what's being tracked on the native iOS level.

# How

- exposed `initialQuery` param via `type RingMuteSwitchStatus = { isMuted: boolean; initialQuery: boolean }` and `type RingMuteSwitchEventCallback = (status: RingMuteSwitchStatus) => void;`:
  - `addSilentListener(callback: RingMuteSwitchEventCallback)`
  - `useSilentSwitch(): RingMuteSwitchStatus`

# API changes

## breaking changes
- `useSilentSwitch` changes return type from `boolean` to `RingMuteSwitchStatus`
- `addSilentListener` changes parameter type from `(status: boolean) => void` to `(status: RingMuteSwitchEventCallback) => void`
- type `eventCallback` is replaced by `RingMuteSwitchEventCallback`
## new features
- introduced `type RingMuteSwitchStatus = { isMuted: boolean; initialQuery: boolean }`
- introduced `type RingMuteSwitchStatusCallback = (status: RingMuteSwitchStatus) => void`

# Side effects / issues - possibly to be extracted to separate PRs

- [ ] I am using `yarn@1.22.18` and it completely messes up the `yarn.lock` files format 🤔 Ideally, I would not change the format at all. What `yarn` version should I use? I've failed to find instructions on that in the repository.
- [ ] I found out that `yarn lint` was not working, because of two errors:
  1. `babel.config.js` imports non-existing file `../package.json`
  2. there's no `babel-plugin-module-resolver` module installed in the root `node_modules`

# Test plan

1. open iOS example app on physical device
2. see the changes in action:

| preview |
|-|
| <video src="https://user-images.githubusercontent.com/16623003/193143477-2d7bcf9d-2f61-4e20-9e95-2e17e583140f.MP4" height=500 /> |
